### ISSUE TITLE
OpenAPI 3.1 support

### ISSUE THREAD

**Issue by letmaik** [lukeautry/tsoa#794]
## Sorting

- **I'm submitting a ...**

  - [ ] bug report
  - [x] feature request
  - [ ] support request

- I confirm that I
  - [x] used the [search](https://github.com/lukeautry/tsoa/search?type=Issues) to make sure that a similar issue hasn't already been submit

## Description

OpenAPI 3.1 is not released yet, but will be soon. It would be nice if tsoa gets 3.1 support eventually. This ticket should track any discussion and issues related to that.

**Comment by Tajnymag** [comment]
This issue should be kept indefinitely until OpenApi 3.1 is implemented. There's no reason to close it.

**Comment by WoH** [comment]
I'll reopen it, but the spec tsoa generates is OpenAPI 3.1 compatible.

**Comment by Tajnymag** [comment]
@WoH Yes, that's correct, the issue might be only tied to the config. At least that's why I came here.

What features are you missing, @letmaik?

**Comment by letmaik** [comment]
@Tajnymag I'm not missing features, but since there are breaking changes with OpenAPI 3.1 I thought it would be worth explicitly checking that tsoa is compatible. See https://github.com/OAI/OpenAPI-Specification/releases/tag/3.1.0-rc0.

An example is `nullable` which has been removed, and a quick search in tsoa turned up this:
https://github.com/lukeautry/tsoa/blob/c19d15d9cad1790544c4d5ace23c46bbad678548/packages/runtime/src/swagger/swagger.ts#L239-L241

**Comment by WoH** [comment]
Seems like this was changed, my last info was deprecated.
Seems like I should revisit this, as they decided to drop SemVer to push out this mess as a 3.1. ("minor breaking changes lulz

**Comment by Tajnymag** [comment]
(commenting to keep the issue open)

**Comment by Tajnymag** [comment]
Shush, StaleBot! 😀

**Comment by mitar** [comment]
I would also love it tsoa could generate a 3.1 version without using anything deprecated. Having proper JSON schema is great. Ideally, those JSON schema definitions could be extracted out into a separate document to be used stand-alone as well.

**Comment by WoH** [comment]
Feel free to submit a PR :)

**Comment by alexellison** [comment]
Is there a plan/timeline/backlog for OpenAPI 3.1 support?

**Comment by WORMSS** [comment]
Hello, just registering my interest in the 3.1 support, so Tuples can be used.

It seems even though my controllers/apis do not use GeoJson but some parts of my code does, tsoa refuses to build and GeoJson uses tuples.

**Comment by brckner** [comment]
Any news on openapi 3.1 support?

**Comment by MathieuRA** [comment]
still no news?

**Comment by jefflembeck** [comment]
Is there a list of openapi 3.1 features that are missing somewhere?

So far the one that stands out to me is variadic tuples. We have `@minItems` but it'd be nice to have the type safety attached.
